### PR TITLE
AEA-1449 tac deploy example

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

tac deploy example

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: examples/tac_deploy/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+PARTICIPANTS_AMOUNT=2
```
Comment (DavidMinarsch) [inline examples/tac_deploy/.env:1]: Let's set this to 100 for cloud deployment

File: examples/tac_deploy/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+RUN apk add --no-cache go
+
+# aea installation
+RUN pip install --upgrade pip
+RUN pip install --upgrade --force-reinstall aea[all]==1.0.0
+
+# directories and aea cli config
+COPY /.aea /home/.aea
+WORKDIR /home/agents
+COPY ./packages /home/agents/packages
```
Comment (DavidMinarsch) [inline examples/tac_deploy/Dockerfile:24]: We don't need this! Let's stick to using the ones from registry only.

Comment (DavidMinarsch) [reply examples/tac_deploy/Dockerfile:24]: Ok I see, dummy registry

File: examples/tac_deploy/.env (hunk 1 above)
Comment (solarw) [reply examples/tac_deploy/.env:1]: done

File: examples/tac_deploy/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+fi
+
+if [ -z  "$INACTIVITY_TIMEOUT" ];
+then
+	INACTIVITY_TIMEOUT=3600
+fi
+
+if [ -z  "$PARTICIPANTS_AMOUNT" ];
+then
+	PARTICIPANTS_AMOUNT=2
```
[the remark this answers is not in the document]
Comment (solarw) [reply examples/tac_deploy/entrypoint.sh:25]: this is set in deployment file

File: examples/tac_deploy/tac-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+         - name: PARTICIPANTS_AMOUNT
+           value: "70"
+         - name: MINUTES_TILL_START
+           value: "10"
+         - name: COMPETITION_TIMEOUT
+           value: "86400"
+         - name: INACTIVITY_TIMEOUT
+           value: "3600"
+         - name: LOG_LEVEL
+           value: "INFO"
```
Comment (DavidMinarsch) [inline examples/tac_deploy/tac-deployment.yaml:32]: In recent deployment, the logs are still at DEBUG level, so not sure what's propagated wrongly where

File: examples/tac_deploy/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	aea issue-certificates
+	json=$(printf '{"delegate_uri": null, "entry_peers": ["%s"], "local_uri": "127.0.0.1:%s", "public_uri": null}' "$PEER" "$port")
+	aea config set --type dict vendor.fetchai.connections.p2p_libp2p.config "$json"
+	log_file=$agent_data_dir/$name.log
+	json=$(printf '{"version": 1, "handlers": {"console": {"class": "logging.StreamHandler", "level": "%s"}, "file": {"class": "logging.FileHandler", "filename": "%s", "mode": "w", "level": "%s"}}, "loggers": {"aea": {"level": "%s", "handlers": ["console", "file"]}}}' "$LOG_LEVEL" "$log_file" "$LOG_LEVEL" "$LOG_LEVEL")
+	aea config set --type dict agent.logging_config "$json"
+	aea config set vendor.fetchai.connections.soef.config.token_storage_path $agent_data_dir/soef_token.txt
+	aea config set agent.skill_exception_policy just_log
+	aea config set agent.connection_exception_policy just_log
+
```
Comment (DavidMinarsch) [inline examples/tac_deploy/entrypoint.sh:87]: Also for the controller please!

Comment (solarw) [reply examples/tac_deploy/entrypoint.sh:87]: it's done for controller as well

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**